Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/storage/KeyCacheRepository.kt

```diff
@@ -138,18 +148,22 @@ class KeyCacheRepository @Inject constructor(
         }
     }
 
-    suspend fun delete(keyInfos: Collection<CachedKeyInfo>) {
+    suspend fun deleteInfoAndFile(keyInfos: Collection<CachedKeyInfo>) {
         Timber.d("delete(keyFiles=%s)", keyInfos)
         keyInfos.forEach { key ->
             getDao().deleteEntry(key)
             Timber.v("Deleted %s", key)
-            val path = getPathForKey(key)
-            if (path.delete()) Timber.v("Deleted cache key file at %s", path)
+            deleteFile(key)
         }
     }
 
+    private fun deleteFile(keyInfo: CachedKeyInfo) {
+        val path = getPathForKey(keyInfo)
+        if (path.delete()) Timber.v("Deleted cache key file at %s", path)
```

Comment:
should catch the error here when deleting? if a file deletion failed for any reason then the others (loop) will continue to be deleted

Reply:
The file may already be deleted from the cache by the OS. We want the others to be deleted anyway